addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,627 @@
+
+        delete_backward_overrides: this.handleDeleteBackward.bind(this),
+        delete_forward_overrides: this.handleDeleteForward.bind(this),
+        shift_tab_overrides: this.handleShiftTab.bind(this),
+        split_element_block_overrides: withSequence(1, this.handleSplitElementBlock.bind(this)),
+        tab_overrides: this.handleTab.bind(this),
+
+        power_buttons_visibility_predicates: this.showPowerButtons.bind(this),
+
+        before_insert_processors: this.handleInsert.bind(this),
+    };
+
+    setup() {
+        this.preventDeleteBackwardContentEnd = false;
+    }
+
+    closestToggleContentInfo(node) {
+        const toggle = closestElement(node, toggleSelector);
+        const title = toggle?.querySelector(titleSelector);
+        const content = toggle?.querySelector(contentSelector);
+        return content?.contains(node) ? { content, title, toggle } : {};
+    }
+
+    closestToggleTitleInfo(node) {
+        const toggle = closestElement(node, toggleSelector);
+        const title = toggle?.querySelector(titleSelector);
+        const content = toggle?.querySelector(contentSelector);
+        return title?.contains(node) ? { content, title, toggle } : {};
+    }
+
+    explodeToggle(toggle) {
+        const title = toggle.querySelector(titleSelector);
+        const content = toggle.querySelector(contentSelector);
+        let contentChildren = children(content);
+        if (contentChildren.length === 1 && isEmptyBlock(contentChildren[0])) {
+            contentChildren = [];
+        }
+        toggle.replaceWith(...children(title), ...contentChildren);
+    }
+
+    forceToggle(toggle, { showContent, restoreSelection } = {}) {
+        toggle.dispatchEvent(
+            new CustomEvent("forceToggle", { detail: { showContent, restoreSelection } })
+        );
+    }
+
+    generateUniqueIds(toggles) {
+        for (const toggle of toggles) {
+            const props = getEmbeddedProps(toggle);
+            props.toggleId = this.getUniqueIdentifier();
+            toggle.dataset.embeddedProps = JSON.stringify(props);
+        }
+    }
+
+    getToggleFromTitleSelection() {
+        const selection = this.dependencies.selection.getEditableSelection();
+        if (!selection.anchorNode) {
+            return;
+        }
+        const { toggle } = this.closestToggleTitleInfo(selection.anchorNode);
+        return toggle;
+    }
+
+    getUniqueIdentifier() {
+        return uuid();
+    }
+
+    /**
+     * Handle all behaviors linked to the use of deleteBackward in the editor:
+     * 1. selection at start of title: explode the toggle and keep title and content as siblings
+     * 2. selection at end of content in a paragraph: unwraps from the toggle content
+     * 3. selection at start of content in a paragraph: merge the paragraph with the title
+     * 4. selection at start of paragraph after a toggle: merge the paragraph at content end
+     */
+    handleDeleteBackward(range) {
```

Comment:
For the sake of efficiency, I did not review any of the specific behaviors below because I assume they are the result of a functional decision in the first place, and they should not alter any behavior outside of your new custom node anyway, since it would break our tests otherwise.